cmake/DeployQt5.cmake

```diff
@@ -0,0 +1,49 @@
+# Based on https://stackoverflow.com/a/41199492
+if(WIN32 AND TARGET Qt5::qmake AND NOT TARGET Qt5::windeployqt)
+    get_target_property(_qt5_qmake_location Qt5::qmake IMPORTED_LOCATION)
+
+    execute_process(
+        COMMAND "${_qt5_qmake_location}" -query QT_INSTALL_PREFIX
+        RESULT_VARIABLE return_code
+        OUTPUT_VARIABLE qt5_install_prefix
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    set(imported_location "${qt5_install_prefix}/bin/windeployqt.exe")
+    if(EXISTS ${imported_location})
+        add_executable(Qt5::windeployqt IMPORTED)
+
+        set_target_properties(
+            Qt5::windeployqt PROPERTIES
+            IMPORTED_LOCATION ${imported_location}
+        )
+    endif()
+endif()
+
+function(plasma_deploy_qt)
+    include(CMakeParseArguments)
+    cmake_parse_arguments(_pdqt "" "TARGET;DESTINATION" "" ${ARGN})
+
+    if(NOT _pdqt_TARGET OR NOT _pdqt_DESTINATION)
+        message(FATAL_ERROR "plasma_deploy_qt must have a valid TARGET and DESTINATION")
+    endif()
+
+    # Save some ruddy characters and install the target
+    install(TARGETS ${_pdqt_TARGET} DESTINATION ${_pdqt_DESTINATION})
+
+    if(TARGET Qt5::windeployqt)
+        # Copy Qt DLLs to the build directory so the stupid application can be run in Visual Studio
+        add_custom_command(
+            TARGET ${_pdqt_TARGET} POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E remove_directory "$<TARGET_FILE_DIR:${_pdqt_TARGET}>/qt"
```

Comment:
If this gets run more than once (which it does), each target will re-populate the deployment directory.  Our Qt apps all use the same Qt libs, so the end result is probably still fine, but this still does more work than necessary.

Perhaps instead we can make a list of executables that need to be considered for windeployqt, and then a single target that runs it on all of them at the same time (windeployqt can take multiple executables on the command line)...

Reply:
I was initially considering doing that by doing cmake's undocumented function overloading... But vcpkg already does that for `add_executable()`, which means we'd introduce an inifite loop. Maybe we could have a `plasma_add_qt_executable` that handles the book-keeping for us? Appending to a list in the CMakeLists.txt or having a hardcoded list seems kind of dirty to me.

Reply:
I don't know why you need to overload anything...  You could just switch to the normal `install` command for the Qt executables, and then run the deployment from the Tools directory's CMakeLists.txt, since that file already has knowledge of the full set of Qt executables to build...

Reply:
I'm not certain how to obtain the list of those executables aside from hardocding a list of them (yuck) or injecting logic to gain it.

Reply:
Another option, which still duplicates work but at least doesn't potentially delete needed files, would be to add a target that only cleans the `qt` staging directory once, and then still have each Qt executable run its own deployment...

Reply:
One thing to note is that I've removed the default specification of `CMAKE_RUNTIME_DIRECTORY`, so, with Visual Studio, each executable is now building in its own directory and has its own qt staging directory. Previously, they were all going into `<builddir>/bin/<config>` So, if you're debugging one of these directly from the IDE, we might not want to clear its staging directory--you'd potentially be blasting the dependencies away before a debug session. I'm not sure what the behavior is with single config generators eg makefiles.